src/backend/codegen/include/codegen/codegen_interface.h

```diff
@@ -0,0 +1,109 @@
+//---------------------------------------------------------------------------
+//  Greenplum Database
+//  Copyright (C) 2016 Pivotal Software, Inc.
+//
+//  @filename:
+//    codegen_interface.h
+//
+//  @doc:
+//    Interface for all code generator
+//
+//---------------------------------------------------------------------------
+#ifndef GPCODEGEN_CODEGEN_INTERFACE_H_  // NOLINT(build/header_guard)
+#define GPCODEGEN_CODEGEN_INTERFACE_H_
+
+#include <string>
+#include <vector>
+
+
+namespace gpcodegen {
+
+/** \addtogroup gpcodegen
+ *  @{
+ */
+
+// Forward declaration
+class CodeGenUtils;
```

Comment:
I would rather it be name as something else - related to LLVM or IR. but if we're going to keep this as it, please rename this to CodegenUtils to match the file: codegen_utils.cc